lang/mozjs78/Portfile

```diff
@@ -0,0 +1,168 @@
+
+if { ${os.platform} eq "darwin" && ${os.major} < 11 } {
+    depends_build-append port:cctools
+    configure.env-append AR=${prefix}/bin/ar
+}
+
+patchfiles          patch-js-config.in.diff
+
+# Use absolute path for install_name
+post-patch {
+    reinplace "s|@executable_path|${prefix}/lib|g" ${worksrcpath}/config/rules.mk
+    reinplace "s|sdk_max_version = Version('10.15.4')|sdk_max_version = Version('12.1')|g" ${worksrcpath}/build/moz.configure/toolchain.configure
+}
+
+configure.perl      /usr/bin/perl
+configure.python    ${prefix}/bin/python2.7
+
+configure.env-append \
+                    SHELL=/bin/bash
+
+configure.dir       ${worksrcpath}/js/src/obj
+configure.cmd       ../configure
+
+# about jemalloc, see
+# https://www.mail-archive.com/dev@couchdb.apache.org/msg47396.html
+configure.args      --with-system-nspr \
+                    --disable-readline \
+                    --disable-jemalloc
+
+configure.universal_args-delete --disable-dependency-tracking
+
+if { ${os.platform} eq "darwin" && ${os.major} < 13 } {
+    configure.args-append   --disable-xcode-checks
+}
+
+build.env-append    SHELL=/bin/bash
+build.dir           ${worksrcpath}/js/src/obj
+destroot.dir        ${worksrcpath}/js/src/obj
+
+post-destroot {
+    # make static lib name version specific to avoid conflict with other mozjs versions
+    move ${destroot}${prefix}/lib/libjs_static.ajs ${destroot}${prefix}/lib/libjs78_static.ajs
+}
+if {![info exists universal_possible]} {
+    set universal_possible [expr {${os.universal_supported} && [llength ${configure.universal_archs}] >= 2}]
+}
+
+if {${universal_possible} && [variant_isset universal]} {
+    set merger_host(x86_64) x86_64-apple-${os.platform}${os.major}
+    set merger_host(i386) i686-apple-${os.platform}${os.major}
+    set merger_configure_args(x86_64) "--host=x86_64-apple-${os.platform}${os.major} --target=x86_64-apple-${os.platform}${os.major}"
+    set merger_configure_args(i386) "--host=i686-apple-${os.platform}${os.major} --target=i686-apple-${os.platform}${os.major}"
+} else {
+    if {${build_arch} eq "i386"} {
+        configure.args-append \
+            --host=i686-apple-${os.platform}${os.major} \
+            --target=i686-apple-${os.platform}${os.major}
+    } elseif {${build_arch} in "x86_64 ppc ppc64"} {
+        configure.args-append \
+            --host=${build_arch}-apple-${os.platform}${os.major} \
+            --target=${build_arch}-apple-${os.platform}${os.major}
+    } elseif {${build_arch} in "arm64"} {
+        configure.args-append \
+            --host=aarch64-apple-${os.platform}${os.major} \
+            --target=aarch64-apple-${os.platform}${os.major}
+        patchfiles-append \
+                    patch-js-src-wasm-WasmSignalHandlers.cpp.diff
+    }
+}
+
+post-configure {
+    reinplace "s|OS_LIBS = |OS_LIBS = -licudata |g" ${worksrcpath}/js/src/obj/config/autoconf.mk
+}
+
+# Now for 10.4/10.5...
+if { ${os.platform} eq "darwin" && ${os.major} < 10 } {
+
```

Comment:
The `rust` build requirement is going to knock out 10.4 and 10.5. This whole block (and associated patches) can be replaced with a `known_fail yes` and error message.

10.6 is also knocked out for now, but rumor has it that @catap is working on a fix...

Reply:
@evanmiller yes, I'm almost finished some rust on macOS 10.6 :)